Source/DafnyCore/Backends/Dafny/DafnyCodeGenerator.cs

```diff
@@ -826,10 +859,10 @@ protected override void EmitNameAndActualTypeArgs(string protectedName, List<Typ
 
       if (GetExprBuilder(wr, out var st) && st.Builder is CallExprBuilder callExpr) {
         var signature = callExpr.Signature;
-        callExpr.SetName((DAST.CallName)DAST.CallName.create_CallName(Sequence<Rune>.UnicodeFromString(protectedName), receiverType, receiverArg, signature));
+        callExpr.SetName((DAST.CallName)DAST.CallName.create_CallName(Sequence<Rune>.UnicodeFromString(protectedName), receiverType, receiverArg, receiverAsArgument, signature));
```

Comment:
Are you afraid arguments could be flipped during maintenance, or that booleans are too vague to be properly maintained?

Reply:
Both really, either case leading to passing the wrong values.

A common pattern is to use enums instead, even if just `{ ENABLE_FOO, DISABLE_FOO }`, because then if you mix up their position you get a type mismatch. Named parameters also helps a lot.

I'm just generally pointing out that the interface between C# and Dafny is particularly risky in this way.

Reply:
I'll keep this in mind while I'll refactor this later on. Thanks.

Reply:
I'm actually already doing that in the next PR which otherwise would have introduced a second boolean. I'd say, as soon as there is a second boolean, there is a need for typed booleans.